### ISSUE TITLE
[BUG/ISSUE] Incorrect units for ProdLoss diagnostic output in GEOS-Chem 13.0.0+

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#965]
@ltmurray wrote:

>The units output by default into the ProdLoss NetCDF diagnostics for the family and individual species production and loss rates have recently changed from molecules cm-3 s-1 to kg s-1. 
>The model documentation indicates that these units should still be molecules cm-3 s-1 and the values themselves appear to still be in those units.
>Was there a conscious attempt to change the units, or is this is a typo introduced into the diagnostics code?

This is a bug in the diagnostics code. The units of the ProdLoss diagnostic are still in molecules cm-3 s-1. In routine `Get_Metadata_State_Diag` in `Headers/state_diag_mod.F90` there is [a check for `isFullChem`](https://github.com/geoschem/geos-chem/blob/ba58da4a6e447e47714fe5e7da488d15432a6fdc/Headers/state_diag_mod.F90#L10964-L10994) using a logical from `Headers/diaglist_mod.F90`. The bug is in the code that determines the value of `isFullChem`:

```
       ! Find out if this is a full-chemistry simulation
       IF ( INDEX( Line, 'Simulation name' ) > 0 ) THEN
          CALL StrSplit( Line, ':', SubStrs, N )
          SELECT CASE( To_UpperCase( ADJUSTL( SubStrs(2) ) ) )
             CASE( 'ACIDUPTAKE', 'APM',              'BENCHMARK',            &
                   'COMPLEXSOA', 'COMPLEXSOA_SVPOA', 'HEMCO',                &
                   'MARINEPOA',  'RRTMG',            'STANDARD',             &
                   'TROPCHEM',   'TOMAS12',          'TOMAS15',              &
                   'TOMAS30',    'TOMAS40'                                  )
                IsFullChem = .TRUE.
             CASE DEFAULT
                IsFullChem = .FALSE.
          END SELECT
       ENDIF
```

In GEOS-Chem 13.3.0, the input files for all full-chemistry simulations were consolidated so that the simulation names listed above were replaced with a single "fullchem" option but this code was not updated so `isFullChem` was always false. To fix the units saved out to the netCDF file, we need to change those lines to:

```
       ! Find out if this is a full-chemistry simulation
       IF ( INDEX( Line, 'Simulation name' ) > 0 ) THEN
          CALL StrSplit( Line, ':', SubStrs, N )
          SELECT CASE( To_UpperCase( ADJUSTL( SubStrs(2) ) ) )
             CASE( 'FULLCHEM' )
                IsFullChem = .TRUE.
             CASE DEFAULT
                IsFullChem = .FALSE.
          END SELECT
       ENDIF
```

This bug impacts ProdLoss diagnostic files saved out by GEOS-Chem 13.3.0-13.3.0. We will fix it in 13.3.1.

**Comment by msulprizio** [comment]
This bug is now fixed in commit https://github.com/geoschem/geos-chem/commit/b1f020693b84efba303ba85ef642af65c69562c4, which will be included in 13.3.1.

**Comment by yantosca** [comment]
Also see PR #978
